src/core/server/metrics/analytics/ops_metrics_event_schema.ts

```diff
@@ -0,0 +1,298 @@
+          },
+          size_limit: {
+            type: 'byte',
+            _meta: { description: 'v8 heap size limit' },
+          },
+        },
+      },
+      resident_set_size_in_bytes: {
+        type: 'byte',
+        _meta: { description: 'node rss' },
+      },
+    },
+    _meta: {
+      description: 'heap memory usage',
+      optional: false,
+    },
+  },
+  event_loop_delay: {
+    type: 'double',
+    _meta: { description: 'mean event loop delay since last collection' },
+  },
+  event_loop_delay_histogram: {
+    properties: {
+      fromTimestamp: {
+        type: 'date',
+        _meta: {
+          description:
+            'The first timestamp the interval timer kicked in for collecting data points.',
+        },
+      },
+      lastUpdatedAt: {
+        type: 'date',
+        _meta: {
+          description: 'Last timestamp the interval timer kicked in for collecting data points.',
+        },
+      },
+      min: {
+        type: 'double',
+        _meta: { description: 'The minimum recorded event loop delay.' },
+      },
+      max: {
+        type: 'long',
+        _meta: { description: 'The maximum recorded event loop delay.' },
+      },
+      mean: {
+        type: 'long',
+        _meta: { description: 'The mean of the recorded event loop delays.' },
+      },
+      exceeds: {
+        type: 'long',
+        _meta: {
+          description:
+            'The number of times the event loop delay exceeded the maximum 1 hour event loop delay threshold.',
+        },
+      },
+      stddev: {
+        type: 'long',
+        _meta: { description: 'The standard deviation of the recorded event loop delays.' },
+      },
+      percentiles: {
+        properties: {
+          '50': {
+            type: 'long',
+            _meta: { description: '50th percentile of delays of the collected data points.' },
+          },
+          '75': {
+            type: 'long',
+            _meta: { description: '75th percentile of delays of the collected data points.' },
+          },
+          '95': {
+            type: 'long',
+            _meta: { description: '95th percentile of delays of the collected data points.' },
+          },
+          '99': {
+            type: 'long',
+            _meta: { description: '99th percentile of delays of the collected data points.' },
+          },
+        },
+        _meta: {
+          description: 'An object detailing the accumulated percentile distribution.',
+          optional: false,
+        },
+      },
+    },
+    _meta: {
+      description: 'node event loop delay histogram since last collection',
+      optional: false,
+    },
+  },
+  uptime_in_millis: {
+    type: 'long',
+    _meta: { description: 'uptime of the kibana process' },
+  },
+};
+
+export const opsMetricsEventSchema: RootSchema<OpsMetricsEvent> = {
+  collected_at: {
+    type: 'date',
+    _meta: { description: 'Time metrics were recorded at.' },
+  },
+  single_process: {
```

Comment:
@afharo I didn't want to map the `processes[0]` entry to the _deprecated_ `process` field, as it's kinda a misrepresentation of what we're doing here. Instead and since the schema is already an abstract representation of what we're sending, I added a custom field here.